src/cluster_slot_stats.c

```diff
@@ -113,6 +119,46 @@ static void addReplyOrderBy(client *c, int order_by, long limit, int desc) {
     addReplySortedSlotStats(c, slot_stats, limit);
 }
 
+/* Resets applicable slot statistics. */
+void clusterSlotStatReset(int slot) {
+    /* key-count is exempt, as it is queried separately through `countKeysInSlot()`. */
+    memset(&server.cluster->slot_stats[slot], 0, sizeof(slotStat));
+}
+
+void clusterSlotStatResetAll(void) {
+    memset(server.cluster->slot_stats, 0, sizeof(server.cluster->slot_stats));
+}
+
+/* For cpu-usec accumulation, nested commands within EXEC, EVAL, FCALL are skipped.
+ * This is due to their unique callstack, where the c->duration for
+ * EXEC, EVAL and FCALL already includes all of its nested commands.
+ * Meaning, the accumulation of cpu-usec for these nested commands
+ * would equate to repeating the same calculation twice.
+ */
+static int canAddCpuDuration(client *c) {
+    return server.cluster_slot_stats_enabled &&  // Config should be enabled.
+           server.cluster_enabled &&             // Cluster mode should be enabled.
+           c->slot != -1 &&                      // Command should be slot specific.
+           (!server.execution_nesting ||         // Either;
+            (server.execution_nesting &&         // 1) Command should not be nested, or
+             c->realcmd->flags & CMD_BLOCKING)); // 2) If command is nested, it must be due to unblocking.
```

Comment:
Yeah, this seems OK to me. I'll still create an issue about execution nesting in blocking but we can resolve that after rc1.

Reply:
https://github.com/valkey-io/valkey/issues/813